### PR TITLE
Fix register deprecation

### PR DESCRIPTION
**Pull request by shtirlic**
Cleanup all register keyword, since it obsolete.
https://en.wikipedia.org/wiki/Register_(keyword)

### CODE REVIEW

File: goldlib/gall/gdefs.h [hunk 1]
```diff
@@ -346,6 +346,14 @@ template <class T> inline bool make_bool_not(T a)
 #define MinV minimum_of_two
 #define MaxV maximum_of_two
 
+/*  ------------------------------------------------------------------ */
+/* rigister deprecated since C++11                                     */
+
+#if __cplusplus > 199711L
+#define register
```
Comment (vitaliy-aksyonov) [inline goldlib/gall/gdefs.h:353]: I'd better just remove that keyword in all places in golded codebase. I don't believe it improves performance at all.

File: goldlib/gall/gdefs.h [hunk 2]
```diff
@@ -346,6 +346,14 @@ template <class T> inline bool make_bool_not(T a)
 #define MinV minimum_of_two
 #define MaxV maximum_of_two
 
+/*  ------------------------------------------------------------------ */
+/* rigister deprecated since C++11                                     */
```
Comment (vitaliy-aksyonov) [inline goldlib/gall/gdefs.h:350]:
```suggestion
/* register deprecated since C++11                                     */
```

**Review by vitaliy-aksyonov — CHANGES_REQUESTED** [review]
Just remove register keyword in code, No need to add this define.

File: goldlib/gall/gdefs.h (hunk 1 above)
Comment (shtirlic) [reply goldlib/gall/gdefs.h:353]: Yeah, will do

**Review by vitaliy-aksyonov — APPROVED** [review]
Looks good to me

**Pull request merged**